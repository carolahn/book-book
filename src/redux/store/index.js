import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import reducers from "../reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(reducers, composeEnhancers(applyMiddleware(thunk)));

export default store;

// import { createStore, applyMiddleware } from "redux";
// import thunk from 'redux-thunk';
// import reducers from '../reducers';

// const store = createStore(reducers, applyMiddleware(thunk));
// export default store;
