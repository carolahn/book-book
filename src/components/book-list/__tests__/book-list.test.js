import React from "react";
import renderer from "react-test-renderer";
import BookList from "../index";

import {shallow} from "enzyme";

import { Provider } from "react-redux";
import store from "../../../redux/store";

import { BrowserRouter, Switch, Route } from "react-router-dom";

const TestFunfa = ({ children }) => (
  <Provider store={store}>
    <BrowserRouter>
      <Switch>
        <Route path="/">{children}</Route>
      </Switch>
    </BrowserRouter>
  </Provider>
);

describe("Render Test", () => {
  it("render empty when there is no book", () => {
    const tree = renderer
      .create(<BookList showBooks={[]} type="timeline" size={1000} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render with books", () => {
    const books = [
      {
        title: "title",
        author: "author",
        shelf: 1,
        image_url: "image_url",
        grade: 0,
        categories: "categories",
        review: "review",
        google_book_id: 1,
        creator: { id: 1, user: "Cassa" },
      },
      {
        title: "title2",
        author: "author2",
        shelf: 1,
        image_url: "image_url2",
        grade: 0,
        categories: "categories2",
        review: "review2",
        google_book_id: 2,
        creator: { id: 1, user: "Cassa" },
      },
    ];

    const tree = renderer
      .create(
        <TestFunfa>
          <BookList showBooks={books} type="timeline" page={1} size={1000} />
        </TestFunfa>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("", () => {
  it("", () => {
    expected().toMatchSnapshot();
  });

  it("", () => {});
});
