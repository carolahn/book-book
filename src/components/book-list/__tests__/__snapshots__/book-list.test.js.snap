// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Test render books 1`] = `
<div
  className="sc-bdnylx dvJgMb"
>
  <div />
</div>
`;

exports[`Render Test render empty when there is no book 1`] = `
<div
  className="sc-bdnylx dvJgMb"
>
  <div />
</div>
`;
